### PR TITLE
Ban trailing whitespace from the entire project, forever

### PR DESCRIPTION
**Pull request by linkmauve**
This removes every trailing whitespace we currently have, and then add a check in travis for added whitespace, before any actual compilation.

### CODE REVIEW

**Comment by neobrain** [comment]
<3

LGTM. Best Citra PR ever. You rock. I mean it! :D

File: .travis-build.sh [hunk 1]
```diff
@@ -3,10 +3,16 @@
 set -e
 set -x
 
+if grep -qr '\s$' src *.yml *.txt *.md Doxyfile .gitignore .gitmodules .travis* dist/*.desktop \
```
Comment (yuriks) [inline .travis-build.sh]: You shouldn't pass `-q` to grep. That way we'll be able to see what files/lines have whitespace.

**Comment by linkmauve** [comment]
Rebased on top of master, fixing yuriks’ note.

**Comment by yuriks** [comment]
LGTM

**Comment by yuriks** [comment]
Shout out to @bunnei for being the first sucka to get caught by this one. :D

**Pull request merged**